docs/syntax.rst

```diff
@@ -43,74 +120,111 @@ The return value will be automatically cast into the hinted type. e.g.,
             return x, y  # Error
 
 
-We also support **template arguments** (see :ref:`template_metaprogramming`) and **external array arguments** (see :ref:`external`) in Taichi kernels.
+Advanced arguments
+******************
 
-.. warning::
+We also support **template arguments** (see :ref:`template_metaprogramming`) and **external array arguments** (see :ref:`external`) in Taichi kernels.  Use ``ti.template()`` or ``ti.ext_arr()`` as their type-hints respectively.
+
+.. note::
 
    When using differentiable programming, there are a few more constraints on kernel structures. See the **Kernel Simplicity Rule** in :ref:`differentiable`.
 
    Also, please do not use kernel return values in differentiable programming, since the return value will not be tracked by automatic differentiation. Instead, store the result into a global variable (e.g. ``loss[None]``).
 
+
 Functions
 ---------
 
-Use ``@ti.func`` to decorate your Taichi functions. These functions are callable only in `Taichi`-scope. Do not call them in `Python`-scopes.
+A Python function decorated by ``@ti.func`` is a **Taichi function**:
 
 .. code-block:: python
 
-   @ti.func
-   def laplacian(t, i, j):
-       return inv_dx2 * (
-           -4 * p[t, i, j] + p[t, i, j - 1] + p[t, i, j + 1] + p[t, i + 1, j] +
-           p[t, i - 1, j])
+    @ti.func
+    def my_func():
+        ...
 
-   @ti.kernel
-   def fdtd(t: ti.i32):
-       for i in range(n_grid): # Parallelized
-           for j in range(n_grid): # Serial loops in each parallel threads
-               laplacian_p = laplacian(t - 2, i, j)
-               laplacian_q = laplacian(t - 1, i, j)
-               p[t, i, j] = 2 * p[t - 1, i, j] + (
-                   c * c * dt * dt + c * alpha * dt) * laplacian_q - p[
-                              t - 2, i, j] - c * alpha * dt * laplacian_p
+    @ti.kernel
+    def my_kern():
+        ...
+        my_func()  # call functions from Taichi-scope
+        ...
 
+    my_kern()      # call kernels from Python-scope
 
-.. warning::
 
-    Functions with multiple ``return`` statements are not supported for now. Use a **local** variable to store the results, so that you end up with only one ``return`` statement:
+Functions are to be called from **Taichi-scope**.
```

Comment:
I think we'd better differentiate the term functions and taichi functions.

Reply:
```suggestion
Taichi functions can only be called from **Taichi-scope**.
```